roles/telemetry_logging/tasks/logging_pod_tests.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+- name: Get Pod Instance "{{ pod_status_str }}" 
+  ansible.builtin.shell:
+    cmd: |
+      oc get pods -n  "{{ nspace }}" | grep "{{ item }}" | grep "{{ pod_status_str }}" | awk '{print $1;}'
+  register: podinstance
+  changed_when: false
+  failed_when: false
+
+- name: Check terminated pod 
+  ansible.builtin.shell:
+    cmd: oc get pod -n "{{ nspace }} {{ podinstance.stdout }}"
+  register: output
+  changed_when: false
+  failed_when: false
+
+- name: Test pod does not exist - "{{ item }}"
+  ansible.builtin.set_fact:
+     exists: FAILED
+
+- name: Test pod exists - "{{ item }}"
+  ansible.builtin.set_fact:
+    exists: PASSED
+  when: 
+    - output.rc == 0
+    - podinstance.stdout != ""
+
+- name: Push results to "{{ pod_out_file }}" 
+  ansible.builtin.shell:
+    'echo "{{ item }},{{ podinstance.stdout }},{{ exists }}" >> "{{ pod_out_file }}"'
```

Comment:
You  should look into a few of the other modules in ansible instead of using shell all the time. 
ansible lint can sometimes be very useful here, since some of the rules include these kind of suggestions.